tests/integration/test_charm.py

```diff
@@ -209,6 +210,91 @@ async def test_deploy_and_relate_postgresql(ops_test: OpsTest):
     )
     assert result["ok"]
 
+    logger.info(f"Unlock (unreleate) {DATA_INTEGRATOR} for the next tests")
+    await ops_test.model.applications[DATA_INTEGRATOR].remove_relation(
+        f"{DATA_INTEGRATOR}:postgresql", f"{POSTGRESQL[ops_test.cloud_name]}:database"
+    )
+
+    # PgBouncer section
+    logger.info(f"Test the relation with {PGBOUNCER[ops_test.cloud_name]}.")
+    num_units = 0 if ops_test.cloud_name == "localhost" else 1
```

Comment:
What is the expected behavior here ? Why the number of units will be zero for the vm charm ?

Reply:
Subordinate charm cannot be installed with num_unit higher then 0. Otherwise:
> FAILED tests/integration/test_charm.py::test_deploy_and_relate_postgresql - juju.errors.JujuError: ['subordinate application must be deployed without units']